likecoin/js/admin/likecoin_metabox.js

```diff
@@ -56,35 +56,128 @@ async function onISCNCallback(event) {
   ISCNTextField.innerHTML = `<a rel="noopener" target="_blank" href="${url}">${status}</a>`;
 }
 
+async function onLikePayCallbackAndUploadArweave(event) {
+  if (event.origin !== 'https://like.co') { // For development, skip this line.
+    return;
+  }
+  try {
+    const { action, data } = JSON.parse(event.data);
+    if (action !== 'TX_SUBMITTED') return;
+    const { tx_hash: txHash, error, success } = data;
+    if (error || success === false) return;
+    const res = await jQuery.ajax({
+      type: 'POST',
+      url: `${wpApiSettings.root}likecoin/v1/posts/${wpApiSettings.postId}/arweave/upload`,
+      dataType: 'json',
+      contentType: 'application/json; charset=UTF-8',
+      data: JSON.stringify({ txHash }), // LIKEpay txHash
+      method: 'POST',
+      beforeSend: (xhr) => {
+        xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+      },
+    });
+    if (!res.data) {
+      throw new Error('UPLOAD_TO_ARWEAVE_ERROR');
+    }
+    if (res.data.arweaveId && res.data.ipfsHash) {
+      onSubmitToISCN();
+    }
+  } catch (error) {
+    console.error(`Error occurs when uploading to Arweave: ${error}`);
+  }
+}
+
 function onSubmitToISCN(e) {
-  e.preventDefault();
+  if (e) e.preventDefault();
   const {
-    title,
-    ipfsHash,
-    tags,
-    url,
+    title, mattersIPFSHash, arweaveIPFSHash, tags, url, arweaveId,
   } = lcPostInfo;
   const { siteurl } = wpApiSettings;
-  if (!ipfsHash) return;
-  const titleString = encodeURIComponent(title);
-  const tagsArray = tags || [];
-  const tagsString = tagsArray.join(',');
-  const urlString = encodeURIComponent(url);
-  const redirectString = encodeURIComponent(siteurl);
-  const likeCoISCNWidget = `https://like.co/in/widget/iscn?fingerprint=${ipfsHash}&publisher=matters&title=${titleString}&tags=${tagsString}&opener=1&url=${urlString}&redirect_uri=${redirectString}`;
-  window.open(likeCoISCNWidget, '_blank', 'menubar=no,location=no,width=576,height=768');
-  window.addEventListener('message', onISCNCallback, false);
+  try {
+    if (!mattersIPFSHash && !arweaveIPFSHash) {
+      throw new Error('NO_IPFS_HASH_FOUND');
+    }
```

Comment:
Although it should not happen (and should be a bug if it occurs), I think we can submit iscn as long as any of these exists: `mattersIPFSHash || arweaveIPFSHash || arweveId`